static/app/components/events/interfaces/performance/spanCountChart.tsx

```diff
@@ -39,18 +40,50 @@ export function SpanCountChart({issue, event, location, organization}: Props) {
   const project = [1];
   const spanOp = event.contexts.performance_issue.op;
 
-  function renderChart(data: HistogramData, _data: HistogramData) {
+  /**
+   * As affected data has larger bins then all data, combines the all data bins to match the affected data bin size.
+   * @param allData echarts histogram data array of all transactions
+   * @param affectedData echarts histrogram data array of affected transactions
+   * @returns echarts histogram data array
+   */
+  function updateBins(allData: HistogramData, affectedData) {
+    const allDataWithNewBins: HistogramData = [];
+    let currentAffectedBinIndex = 0;
+    let currentBinIndex = 0;
+    while (
+      currentAffectedBinIndex < affectedData.length &&
+      currentBinIndex < allData.length
+    ) {
+      const currentBinMax = affectedData[currentAffectedBinIndex].bin;
+      let binCount = 0;
+      while (
+        currentBinIndex < allData.length &&
+        allData[currentBinIndex].bin <= currentBinMax
+      ) {
+        binCount += allData[currentBinIndex].count;
+        currentBinIndex++;
+      }
+      allDataWithNewBins.push({bin: currentBinMax, count: binCount});
+      currentAffectedBinIndex++;
+    }
+    return allDataWithNewBins;
+  }
```

Comment:
some tests would be great here to make sure it works as intended. if we ever end up changing this, it's good to have a test that would check that nothing accidentally broke :)

Reply:
I was actually just in the middle of adding them :)

Reply:
I also moved this code into a shared util, as it could be useful for future stacked histograms!